runner-plugins/runner-scribe/main.go

```diff
@@ -209,6 +256,94 @@ func makeVulnerability(initems []gozdef.VulnEvent, cmd mig.Command) (items []goz
 	return
 }
 
+// given config for an API behind Auth0 (including client ID and Secret), 
+// return an Auth0 access token beginning with "Bearer "
+// pattern from https://auth0.com/docs/api-auth/tutorials/client-credentials
+func GetAuthToken(api ServiceApi) (authToken string) {
+	payload := strings.NewReader("{\"grant_type\":\"client_credentials\",\"client_id\": \"" + api.ClientID + "\",\"client_secret\": \"" + api.ClientSecret + "\",\"audience\": \"" + api.URL + "\"}")
+	req, _ := http.NewRequest("POST", api.AuthEndpoint, payload)
+	req.Header.Add("content-type", "application/json")
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		panic(err)
+	}
+
+	defer res.Body.Close()
+	bodyJSON, _ := ioutil.ReadAll(res.Body)
+	
+	// unpack the JSON into an Auth0 token struct
+	var body Auth0Token
+	err = json.Unmarshal(bodyJSON, &body)
+	if err != nil {
+		panic(err)
+	}
+
+	// serviceAPI expects the Access token in the form of "Bearer <token>"
+	authToken = "Bearer " + body.Access_token
+	return
+}
+
+// query a ServiceAPI instance for the set of all assets
+// load them into a searchable map, keyed to asset hostname
+// the ServiceAPI object must already be loaded with a Bearer token
+func GetAssets(m map[string]ServiceApiAsset, api ServiceApi) (err error){
+	
+	// get json array of assets from serviceapi
+	requestURL := api.URL + "api/v1/assets/"
+	req, err := http.NewRequest(http.MethodGet, requestURL, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Add("accept", "application/json")
+	req.Header.Add("Authorization", api.Token)
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	// unpack the HTTP request response
+	defer res.Body.Close()
+	body, readErr := ioutil.ReadAll(res.Body)
+	if readErr != nil {
+		return readErr
+	}
+
+	// because of the way that ServiceAPI returns the JSON content,
+	// we need to Unmarshal it twice
+	var allAssetsJson string
+	err  = json.Unmarshal(body, &allAssetsJson)
+	if err != nil {
+		return err
+	}
+
+	// convert json into array of ServiceApiAsset objects
+	var allAssets []ServiceApiAsset
+	err  = json.Unmarshal([]byte(allAssetsJson), &allAssets)
+	if err != nil {
+		return err
+	}
+
+	// build a searchable map, keyed on Asset_identifier (which is usually hostname)
+	for _, tempAsset := range allAssets {
+		permanentAsset := tempAsset				//not sure if this is needed
+		m[tempAsset.Asset_identifier] = permanentAsset
+	}
+
+	return
+}
+
+// return the operator and team for a given hostname, provided they are in the map of 
+// ServiceApiAssets. If they are not in the map or if the values are not present, 
+// operator and/or team will return as an empty string ""
+func LookupOperatorTeam(hostname string, m map[string]ServiceApiAsset) (operator string, team string) {
+	
+	operator = m[hostname].Operator
```

Comment:
Is it possible that we may lookup a `hostname` that isn't in the map?  Since we have this abstracted under a function anyway, I propose we make the interface a little more friendly.  Perhaps something like the following.

```go
type Team struct {
  Name string
  Operator string
}

func LookupOperatorTeam(
  hostname string,
  mapping map[string]ServiceApiAsset,
) (Team, bool) {
  asset, found = mapping[hostname]
  if !found {
    return Team{}, false
  }
  team := Team{
    Name: asset.Team,
    Operator: asset.Operator,
  }
  return team, true
}

Reply:
In testing, if hostname is not found in the map, Operator and Team return as empty strings. This is expected behavior and we'll just set them as "" in the mozdef event. Is there anything useful we could do with the returned bool?